tpm2tools/import.go

```diff
@@ -6,44 +6,82 @@ import (
 
 	tpmpb "github.com/google/go-tpm-tools/proto"
 	"github.com/google/go-tpm/tpm2"
+	"github.com/google/go-tpm/tpmutil"
 )
 
 // Import decrypts the secret contained in an encoded import request.
 // The key used must be an encryption key (signing keys cannot be used).
 // The req parameter should come from server.CreateImportBlob.
 func (k *Key) Import(rw io.ReadWriter, blob *tpmpb.ImportBlob) ([]byte, error) {
-	auth, err := k.session.Auth()
+	handle, err := loadHandle(k, rw, blob)
 	if err != nil {
 		return nil, err
 	}
-	private, err := tpm2.Import(rw, k.Handle(), auth, blob.PublicArea, blob.Duplicate, blob.EncryptedSeed, nil, nil)
+	defer tpm2.FlushContext(rw, handle)
+
+	unsealSession, err := newPCRSession(rw, PCRSelection(blob.Pcrs))
 	if err != nil {
-		return nil, fmt.Errorf("import failed: %s", err)
+		return nil, err
 	}
+	defer unsealSession.Close()
 
-	auth, err = k.session.Auth()
+	auth, err := unsealSession.Auth()
 	if err != nil {
 		return nil, err
 	}
-	handle, _, err := tpm2.LoadUsingAuth(rw, k.Handle(), auth, blob.PublicArea, private)
+	out, err := tpm2.UnsealWithSession(rw, auth.Session, handle, "")
 	if err != nil {
-		return nil, fmt.Errorf("load failed: %s", err)
+		return nil, fmt.Errorf("unseal failed: %s", err)
 	}
-	defer tpm2.FlushContext(rw, handle)
+	return out, nil
+}
 
-	unsealSession, err := newPCRSession(rw, PCRSelection(blob.Pcrs))
+func (k *Key) ImportSigningKey(rw io.ReadWriter, blob *tpmpb.ImportBlob) (*Key, error) {
```

Comment:
We don't need the `rw io.ReadWriter` parameter. We can just use `k.rw` as part of the key.

Yes, I realize we had it as part of `Import`, but that was a mistake.

Reply:
I'll remove the the param from Import as part of a breaking change.